pkg/wipeout/wipeout.go

```diff
@@ -68,35 +68,55 @@ func Wipeout(ctx context.Context, client connection.RegistryClient, projectID st
 
 func wipeoutArtifacts(ctx context.Context, client connection.RegistryClient, taskQueue chan<- core.Task, parent string) {
 	it := client.ListArtifacts(ctx, &rpc.ListArtifactsRequest{Parent: parent})
+	names := make([]string, 0)
 	for artifact, err := it.Next(); err == nil; artifact, err = it.Next() {
-		taskQueue <- NewDeleteArtifactTask(client, artifact.Name)
```

Comment:
@seaneganx So as it turns out, deleting resources while iterating through them does seem to be a bad idea. Since the queue is already running, just putting these tasks in the queue is enough to start deletions, and I found that breaking this into two parts fixed the flaky wipeout behavior that was breaking the tests... when I added more logging I found that API deletions would fail frequently (but not always) with a varying number of deployments as child resources. Previously I had also seen version deletions fail. This might have been caused by the iterators skipping resources (Suppose we get 5 resources at a time and delete the first five, then ask for the "next" five. Are those next five the original 6-10 or are they 11-15? This will depend on how the cursor and queries are implemented, but from [this](https://github.com/apigee/registry/blob/2c94ac7bf013279f5c9747c2b6bc37587181a189/server/registry/internal/storage/list.go#L177) I think we'll be getting 11-15 and skipping 6-10)

Reply:
Right. Under the hood our page tokens include an offset into the collection, pointing at where to start the next page. If resources that have already been listed are deleted, then some resources that haven't been listed yet will be skipped.

For your example, the "next" five will be the original 11-15 because the offset for the second page (encoded into the page token) is still 5, but resources 6-15 all shifted 5 places backward after the deletion of resources 1-5.